packages/wdio-cli/src/commands/run.ts

```diff
@@ -194,10 +196,33 @@ export async function handler(argv: RunCommandArguments) {
             }
         })
 
+        const executeKill = () => {
+            removeSignalListeners()
+            if (tsProcess.pid) {
+                // gracefully kill child process
+                kill(tsProcess.pid, 'SIGINT')
+            }
+        }
+        const removeSignalListeners = () => {
+            process.removeListener('SIGINT', executeKill)
+            process.removeListener('SIGTERM', executeKill)
```

Comment:
This would kill the TS process with `SIGINT` rather `SIGTERM` .. I am not sure if this matters a lot

Reply:
I'm considering using async-exit-hook, or another exit-hook, but haven't found one that passes the exit code to the hook. Do you know of any?

Reply:
Maybe it is worth considering a different way to do the whole initial config parsing, maybe through a web-worker that passes down the config object. However this seems problematic as we can't serialize e.g. `onPrepare` hooks through that.

Compared to `v7` we are now running into this because compiling TS with `ts-node` in ESM environment requires passing in these `NODE_OPTIONS`.

Reply:
I ended up forking `exit-hook` and modified it to pass the exit code in the callback handler: https://www.npmjs.com/package/@scg82/exit-hook. Would you like me to use this here?

Reply:
What is the added value using your `exit-hook` fork? If the current solution solves the issue I would prefer that as it doesn't require us to maintain another dependency.